test/controllers/traces_controller_test.rb

```diff
@@ -8,51 +8,27 @@ def test_routes
       { :path => "/traces", :method => :get },
       { :controller => "traces", :action => "index" }
     )
-    assert_routing(
-      { :path => "/traces/page/1", :method => :get },
-      { :controller => "traces", :action => "index", :page => "1" }
-    )
```

Comment:
We should keep some some redirect tests to make sure the redirects aren't accidentally refactored away in future.

Reply:
I'd love to test the redirects but I couldn't find a way to do so... I think I've been here before and failed to find a way to do it :-(

Reply:
We use this approach elsewhere in controller tests:

```
visit foo
assert_redirected_to bar
```

Would that work for here?

Reply:
Oh that might work sure, what I was looking for was away to do it with assert_routing/assert_recognizes rather than going through the whole stack.

Reply:
Yeah, I think since there's no controller involved in a redirect, it's not going to work with those assertions. https://stackoverflow.com/a/10844398/105451 has a reasonably well researched take on this, to do with how the routes are handled differently internally to rails when it's a redirect.